### ISSUE TITLE
Text Documents are altered when merged

### ISSUE THREAD

**Issue by Phiwatec** [py-pdf/PyPDF2#1058]
When using the PdfMerger to merge (append) two simple pdf files it merges them incorrectly.
The first is appended without a problem. But the second one is a mixture of both. If the pdfs are dissimilar this does not happen.
This happens both with pdfs created from LibreOffice Writer and an online convert Service.

```bash
$ python -m platform
Linux-5.10.0-13-amd64-x86_64-with-glibc2.31

$ python -c "import PyPDF2;print(PyPDF2.__version__)"
2.4.1

$ python --version
Python 3.9.2
```

This is a minimal, complete example that shows the issue:

```python
#Basic example from docs
from PyPDF2 import PdfFileMerger, PdfFileReader
merger = PdfFileMerger()
merger.append("LO_first.pdf")
merger.append("LO_second.pdf")
merger.write("LO_out.pdf")
merger.close()


```
There a six attached files:
[LO_first.pdf](https://github.com/py-pdf/PyPDF2/files/9042010/LO_first.pdf) and [LO_second.pdf](https://github.com/py-pdf/PyPDF2/files/9042009/LO_second.pdf) are files created with Libreoffice Writer.
[LO_out.pdf](https://github.com/py-pdf/PyPDF2/files/9042007/LO_out.pdf) is the resulting wrong file
[online_first.pdf](https://github.com/py-pdf/PyPDF2/files/9042012/online_first.pdf) and [online_second.pdf](https://github.com/py-pdf/PyPDF2/files/9042011/online_second.pdf) are created using an online convert service from a plaintext file
[online_out.pdf](https://github.com/py-pdf/PyPDF2/files/9042008/online_out.pdf) is the resulting wrong file

When using Firefox two view the resulting document it does not show a hidden character:
![firefox](https://user-images.githubusercontent.com/52720981/177216719-51ad7021-1430-45fe-8f0a-e1f2f092f739.png)
When using Chrome or Okular it shows a non printable character:
![chrome](https://user-images.githubusercontent.com/52720981/177216721-3d2ba79f-eb21-4827-8b35-a1e7f979885b.png)

In both cases it should be bcde and not abc.

If the files contain very similar content ( "test1", "test2", "test3",etc.) the first file is used for all pages.

Thanks in advance for looking at this behavior.

**Comment by MartinThoma** [comment]
Thank you for reporting the issue. I'll have a look this week (and I hope @MasterOdin / @pubpub-zz have some time as well 😅)

**Comment by Phiwatec** [comment]
Thank you :)

**Comment by pubpub-zz** [comment]
I think the issue is due to the fact that the two initial pdf files are identifying a font named /F1 which has a different definitions in the two files. the first one only refers 3 characters "a" "b" "c" associated with coding 01 02 03 whereas in second find 01 02 03 04 are associated with "b" "c" "d" "e"
after merging the page 2 refers to the same font object as on page 1, which is inducing the incorrect data.
this is a bug confirmed

**Comment by MartinThoma** [comment]
@Phiwatec This issue was likely the same as https://github.com/py-pdf/PyPDF2/issues/1062. It was introduced in 2.4.1 via #207. It was fixed in 2.4.2 (released moments ago) via #1063

Could you please check if things work with `PyPDF2==2.4.2` for you again?
